state_machine/fsm.go

```diff
@@ -0,0 +1,34 @@
+package state_machine
+
+import (
+	"github.com/looplab/fsm"
+)
+
+// NewNodeFSM returns a KISS Finite State Machine that is meant to mimick the various "states" of the node.
+//
+// source: consensus/doc/PROTOCOL_STATE_SYNC.md + additions for P2P
+// The initial implementation is going to be used to understand, from a P2P perspective, if the node requires bootstrapping
+func NewNodeFSM(callbacks *fsm.Callbacks, options ...func(*fsm.FSM)) *fsm.FSM {
+	var cb = fsm.Callbacks{}
+	if callbacks != nil {
+		cb = *callbacks
+	}
+
+	stateMachine := fsm.NewFSM(
+		"stopped",
+		fsm.Events{
+			{Name: "start", Src: []string{"stopped"}, Dst: "P2P_bootstrapping"},
```

Comment:
Can we move these into an iota enum?

I want a single place where I can go and see:
- All the states the node can be in
- All the events we can send around